docs/high-order-components.md

```diff
@@ -0,0 +1,89 @@
+# High Order Components
```

Comment:
These docs are cool. Thanks. I'm a little worried that they will fall out of sync with the code, though. They're fine for now but maybe we can consider something later like [Sphinx for JavaScript](https://github.com/erikrose/sphinx-js). I also really like how easy it is to interlink docs with Sphinx. Hmm. I can't tell how well Sphinx supports Markdown these days, though.